### ISSUE TITLE
Adding Feature Selection for every classifier in the hierarchy

### ISSUE THREAD

**Issue by SarikaKV** [globality-corp/sklearn-hierarchical-classification#17]
Hi,

Thanks for the amazing work!

My requirement is to add feature selection before every classifier step. If it is a lcpn, I need feature selection before every parent node classifier. I am trying to add feature selection before running the classifier. 
I see you provided a function _select_features() to implement in the subclass. I created my own subclass inheriting HierarchicalClassifier. I tried two ways but failed.

(1) Tried modifying _select_features()
It throws inconsistent shapes error since you pass the full X for every node. I tried to use the indices that are passed in _build_features() to extract only the required X. But it fails later at the classifier step giving inconsistent shapes error.
(2) Modifying the classifier function
I also tried modifying the classifier function, _train_local_classifier() directly. Just before you run the classifier, I added my feature selection. It worked fine for the training data. But as anticipated it failed on test data (y_pred = clf.predict(X_test))  since it is not feature selected. I tried avoiding feature selection at the root node so that train and test data would be the same shape but it threw an error at the first feature selected node. 

Can you post an example with some feature selection technique implementation?  

Thanks,
Sarika.

**Comment by adamhadani** [comment]
Hey Sarika,
Sounds like a good feature no pun intended!
Will take a look at this and see if we can address in the near future. If you want to try your hand at a PR, always welcome..

**Comment by fbenites** [comment]
Hi
I am also on that, but have problems, since multi-label.
My approach is to make a pipeline and change the checks for X matrix to something else.
I will make a pull-request if I manage to achieve something.

I will create another issue for the multi-label which is blocking it atm.

Cheers,
Fernando

**Comment by fbenites** [comment]
Partially solved by https://github.com/globality-corp/sklearn-hierarchical-classification/pull/20

**Comment by adamhadani** [comment]
Tracking via PR #33  going forward - will update once merged
